tool/teleport/common/teleport.go

```diff
@@ -195,6 +195,9 @@ func Run(options Options) (app *kingpin.Application, executedCommand string, con
 	start.Flag("db-aws-region",
 		"AWS region AWS hosted database instance is running in.").Hidden().
 		StringVar(&ccf.DatabaseAWSRegion)
+	start.Flag("disable-debug-service",
+		"Disables debug service.").Hidden().
+		BoolVar(&ccf.DisabledDebugService)
```

Comment:
I would name the flag `debug-service`, make it a `BoolVar()` and `.Default(true)`.

This will automatically generate a `--no-debug-service` option that can be used to disable it.

Reply:
Wouldn't adding such a flag make the configuration obsolete? Regarding configuration parsing, the command-line flags have priority over the default and user-provided configurations. For example, given that the value would be `true`, even when users change it to `debug_service.enabled`, it would always get overwritten by the CLI value.

We could name it the `--no-debug-service` (without creating the `--debug-service`) flag. That way, we can keep the configuration in the file so it won't change the current behavior. What do you think?

Reply:
I'd like to avoid the double negatives if we can. For example, `tsh ssh --help` shows this:

```
      --[no-]no-resume           Disable SSH connection resumption
```

`--no-no-resume` is a very awkward UX.

Reply:
> Regarding configuration parsing, the command-line flags have priority over the default and user-provided configurations. For example, given that the value would be `true`, even when users change it to `debug_service.enabled`, it would always get overwritten by the CLI value.

Ah, okay, I see this now. We cannot have `BoolVar` with `Default(true)` because it'd always take precedence over whatever is in the config file and the only way to turn off the debug service would be to pass the `--no-debug-service` flag.

This feels cursed, but after thinking about it for a sec, this behavior makes sense from kingpin's PoV – you say you have a boolean flag and you want the default to be `true`, even if the flag wasn't specified.

So ultimately wouldn't the solution be to have a bool flag called `debug-service` with no default on the flag? Would it then work as expected? As in:

* No flag specified and no config: `debug_service.enabled` through setting defaults on the config.
* No flag and config: `debug_service.enabled` set to whatever is in the config.
* Flag and config: the flag takes precedence over the config.

> We could name it the `--no-debug-service` (without creating the `--debug-service`) flag. That way, we can keep the configuration in the file so it won't change the current behavior. What do you think?

Looking at the output of `teleport start --help`, I think we definitely should go with `--no-debug-service`. There's no other flag that starts with `disable`. I also don't really see why it would need to be hidden.

Reply:
The thing is, we don't have a way to know whether the flag was set to `false` by the user or if it uses the field struct bool default value. So, if you perform a `teleport start` without any flag and no default value to `debug-service`, it would be set to `false`.

Reply:
It feels like something we should address at some point, otherwise I guess we'll run into more and more situations like this one.

What about going with `no-debug-service` without supporting `debug-service` as you suggested in https://github.com/gravitational/teleport/pull/46032#discussion_r1736822939? I feel like we should try to make the flag public if possible.

Reply:
I've updated the name to `no-debug-service` and made it public. I agree the double negative (`--[no-]no-debug-service`) is not good, but it seems like having a flag (e.g., `--debug-service`) that does not affect its standard format (non-negative) is worse. Let me know what you think.